tensorflow/python/eager/backprop.py

```diff
@@ -812,6 +812,10 @@ def watch(self, tensor):
       tensor: a Tensor or list of Tensors.
     """
     for t in nest.flatten(tensor):
+      if not t.dtype.is_floating:
+        logging.vlog(
+            logging.WARN, "The dtype of the watched tensor must be "
+            "floating (e.g. tf.float32), got {!r}".format(t.dtype))
```

Comment:
Sorry, one more nitpick: could you replace vlog(..., "... {!r}".format(t.dtype)) with vlog(..., "... %r", t.dtype) so that interpolation is done lazily? Same for other messages.

Reply:
Done now, Thanks!